paddle/fluid/operators/distributed/communicator.cc

```diff
@@ -83,6 +83,47 @@ Communicator::Communicator(const RpcCtxMap &send_varname_to_ctx,
   recv_threadpool_.reset(new ::ThreadPool(FLAGS_communicator_thread_pool_size));
 }
 
+Communicator::Communicator(const paddle::framework::ProgramDesc &program,
+                           Scope *param_scope) {
+  using RpcCtxMap = operators::distributed::RpcCtxMap;
+  VLOG(3) << "ProcessGraph";
+  RpcCtxMap send_varname_to_ctx;
+  RpcCtxMap recv_varname_to_ctx;
+  for (auto *op : program.Block(0).AllOps()) {
+    VLOG(3) << "node name " << op->Type();
+    if (op->Type() == "send") {
```

Comment:
Can I use communicator beyond send_op and recv_op?

Reply:
currently it can only be used inside send and recv op.